pandas/io/parsers.py

```diff
@@ -343,6 +343,45 @@
 Examples
 --------
 >>> pd.{func_name}('data.csv')  # doctest: +SKIP
+
+Occasionally, you might run into a data file that begins with comments.
+
+data.csv -
+# This file contains info on Apollo space missions.
+# Each row contains info on the mission title, launch date, and duration.
+duration,launch_date,mission
+NaN,21 February 1967,Apollo 1
+10d 20h 09m 03s,11 October 1968,Apollo 7
+06d 03h 00m 42s,21 December 1968,Apollo 8
+...
+
+While this will throw off pd.{func_name}, it can be handled with some minor
+pre-processing.
+
+>>> with open('data.csv', 'r') as f: # doctest: +SKIP
+        comments = []
+        while True:
+            line = f.readline().strip() # Read a single line.
+
+            if line.startswith('#'):
+                comments.append(line)
+            else:
+                break
+
+        columns = line.split(',') # The last line read was the header.
+        df = pd.read_csv(f, header=None) # Pass the file handler to {func_name}
```

Comment:
you can pass names=columns

can you put the comments on prior lines

Reply:
Passed names=columns.

Reply:
Comments are now on prior lines.

Reply:
Good call on the formatting check. It looked quite ghastly for a moment there. Here's what it looks like now.

<img width="660" alt="Screen Shot 2019-04-18 at 10 28 32 PM" src="https://user-images.githubusercontent.com/17144735/56401976-6b9ad080-6229-11e9-9be5-65a5e7deb7e9.png">

Reply:
@jreback Does that work?